File: Backend/Trivagogoro_Backend/Trivagogoro_Backend/Services/DTO/GetFollowedPostDTO.cs
using System;
namespace Trivagogoro_Backend.Services.DTO
{
    public class GetFollowedPostDTO
    {
        // for followed user
        public int flId { get; set; }
        public string flName { get; set; } = "";

        // for the post
        public int fwpId { get; set; }
        public string fwpDescription { get; set; } = "";
        public string fwpTitle { get; set; } = "";
        public int fwpArchivedNum { get; set; } = 0;
        public PostType fwpType { get; set; } = PostType.Rest;
        public int fwpSourceId { get; set; } // Rest => favoriteId, FoodMap => foodmapId

        // for restaurant
        public string placeId { get; set; } = "";
        public List<string> images { get; set; } = new List<string>();
    }
}

File: Frontend/Trivagogoro_Frontend/src/app/models/Responses/GetFollowedPostDTO.ts
export interface GetFollowPostsRes
{
  followedPostDTOs: GetFollowPostsDTO[]
}

export interface GetFollowPostsDTO
{
  flId: number,
  flName: string,
  fwpId: number,
  fwpDescription: string
  fwpTitle: string,
  fwpArchivedNum: number
  fwpType: number,
  fwpSourceId: number,
  placeId: string,
  images: string[]
}
